src/GSoft.Extensions.MediatR.Tests/InMemoryLoggerTracker.cs

```diff
@@ -53,11 +56,15 @@ public void AssertRequestFailed(string startsWith)
         lock (this._logs)
         {
             Assert.Equal(2, this._logs.Count);
-            Assert.Single(this._logs, x => x.StartsWith(startsWith + " started"));
-            Assert.Single(this._logs, x => x.StartsWith(startsWith + " failed after"));
+            Assert.StartsWith(startsWith + " started", this._logs[0].Text);
+            Assert.StartsWith(startsWith + " failed after", this._logs[1].Text);
+            Assert.NotNull(this._logs[0].ActivityId);
+            Assert.Equal(this._logs[0].ActivityId, this._logs[1].ActivityId);
         }
     }
 
+    private sealed record LogRecord(string Text, string? ActivityId);
```

Comment:
There was a bug, we now make sure that the OpenTelemetry span (activity) is the same for both start and end logs.